### DISCUSSION TITLE
trying to get remark frontmatter 2 and remark 12 to work

### DISCUSSION THREAD

**Discussion by shedali** [remarkjs/remark#750] [Q&A] [answered]
following the readme instructions but the frontmatter plugin does not appear to work for me. I'm sure it's something I'm doing wrong with the mismatched versions.

repro here https://stackblitz.com/edit/node-w3zsuy?file=index.js

**Comment by ChristianMurphy** [comment, marked as answer]
remark parse and remark stringify need to be version 8 to work with remark frontmatter 2
https://codesandbox.io/s/remark-debug-forked-6jf6g?file=/src/index.js

**Reply by shedali** [reply]
am I right that appears to return the origin input in the results, i.e. not transformed. My expectation was that the yaml block would be removed from the output. I get the same thing locally, but I think before it stripped out the block.

**Reply by ChristianMurphy** [reply]
> am I right that appears to return the origin input in the results

Yes, remark and it's extension parsers by default allow roundtripping.
The markdown structure that comes in, will be the same as the structure of markdown that is output.

Adding Plugins/Transformers allows changing the content along the way.

`remark-frontmatter` is a parser/stringifier, it doesn't do any changes to the content.
You can transform it with something like https://github.com/mrzmmr/remark-extract-frontmatter
